### PR TITLE
Add spec conformance tests for xml-template-expr

### PR DESCRIPTION
**Pull request by gimantha**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/32635

### CODE REVIEW

File: conformance/lang/expressions/xml-template-expr.balt [hunk 1]
```diff
@@ -0,0 +1,838 @@
...
+    int|float|boolean|string|decimal someValue = true;
+    xml content = xml `<b:BOOK xmlns:b="${someValue}"></b:BOOK>`; // @error xml template expressions are not allowed in namespace attribute values
+    content = xml `<b:BOOK xmlns:b="prefix${someValue}suffix"></b:BOOK>`; // @error xml template expressions are not allowed in namespace attribute values
+}
+
+Test-Case: output
+Description: Test the same xml namespace with different namespace prefix inside the interpolated expression.
+Labels: xml-template-expr, xml
+
+import ballerina/io;
```
Comment (lochana-chathura) [inline conformance/lang/expressions/xml-template-expr.balt:15]: No need add `import ballerina/io;` here and other places

File: conformance/lang/expressions/xml-template-expr.balt [hunk 2]
```diff
@@ -0,0 +1,838 @@
...
+}
+
+Test-Case: error
+Description: Test usage of interpolated expressions of nil type in xml text.
+Labels: xml-template-expr, xml, xml:Text
+
+public function main() {
+    int? firstExpr = 1;
+    xml<xml:Text> x4 = xml `text${firstExpr}`; // @error int? type not allowed in xml text
+    xml<xml:Text> x4 = xml `text${()}`; // @error int? type not allowed in xml text
```
Comment (lochana-chathura) [inline conformance/lang/expressions/xml-template-expr.balt:77]:
```suggestion
    xml<xml:Text> _ = xml `text${firstExpr}`; // @error int? type not allowed in xml text
    xml<xml:Text> _ = xml `text${()}`; // @error int? type not allowed in xml text
```

Need to use wildcard-bindin-pattern for unused variables.

File: conformance/lang/expressions/xml-template-expr.balt [hunk 3]
```diff
@@ -0,0 +1,838 @@
...
+import ballerina/io;
+public function main() {
+    xml content1 = xml `<x:BOOK xmlns:x="http://www.w3.org/TR/html4/}"></x:BOOK>`;
+    xml content2 = xml `<b:BOOKs xmlns:b="http://www.w3.org/TR/html4/}">${content1}</b:BOOKs>`;
+    io:println(content2); // @output <b:BOOKs xmlns:b="http://www.w3.org/TR/html4/}"><x:BOOK xmlns:x="http://www.w3.org/TR/html4/}"/></b:BOOKs>
+}
+
+Test-Case: error
+Description: Test the static type of the interpolated expressions in an attribute value not to be nil.
+Labels: xml-template-expr, xml, raw-template-expr, nil-type
```
Comment (lochana-chathura) [inline conformance/lang/expressions/xml-template-expr.balt:24]: Missing labels union-type, float, decimal etc in here and other places.

**Comment by hasithaa** [comment]
The Team is busy with finalizing the SwanLake GA release. Hence closing this PR temporarily.

File: conformance/lang/expressions/xml_template_expr.balt [hunk 4]
```diff
@@ -0,0 +1,765 @@
...
+
+    xml x = xml `<Books>${xml `<Book>"${book?.coAuthor}"</Book>`}</Books>`; // @error static type of the interpolated expressions in xml content should not be nil
+    xml<xml:Comment> y = xml `<!--${book?.coAuthor}-->`; // @error static type of the interpolated expressions in comments should not be nil
+    xml z = xml `<Books>${xml `<Book author="${book?.coAuthor}">"${book.name}"</Book>`}</Books>`; // @error static type of the expressions in attribute values should not be nil
+    xml<xml:Text> text = xml `${book?.coAuthor}`; // @error static type of the interpolated expressions in xml text should not be nil
+}
+
+Test-Case: error
+Description: Test usage of member access expressions as interpolated expressions in xml content.
+Labels: xml-template-expr, xml, member-access-expr
```
Comment (suleka96) [inline conformance/lang/expressions/xml_template_expr.balt:410]: Shall we add the missing labels here and please check other places in the file as well, there seems to be quite a bit that has been missed.

File: conformance/lang/expressions/xml_template_expr.balt [hunk 5]
```diff
@@ -0,0 +1,765 @@
...
+    xml z = xml `<Books>${xml `<Book author="${book.author}">"${book.name}"</Book>`}</Books>`;
+    io:println(z); // @output <Books><Book author="James">"Ballerina"</Book></Books>
+
+    xml<xml:Text> text = xml `${book.author}`;
+    io:println(text); // @output James
+}
+
+Test-Case: error
+Description: Test usage of optional field access expressions as interpolated expressions in xml content.
+Labels: xml-template-expr, xml, optional-field-access-expr
```
Comment (suleka96) [inline conformance/lang/expressions/xml_template_expr.balt:390]: Missing label `record-type`

File: conformance/lang/expressions/xml_template_expr.balt [hunk 6]
```diff
@@ -0,0 +1,765 @@
...
+    io:println(x); // @output <Books><Book>"34"</Book></Books>
+
+    x = xml `<Books>${xml `<Book>"${firstExpr}"</Book>`}</Books>`;
+    io:println(x); // @output <Books><Book>"34"</Book></Books>
+}
+
+Test-Case: output
+Description: Test usage of queries as interpolated expressions in xml content.
+Fail-Issue: ballerina-platform/ballerina-lang#33076
+Labels: xml-template-expr, xml, query-expr
```
Comment (suleka96) [inline conformance/lang/expressions/xml_template_expr.balt:327]: `query-expr` label is missing

File: conformance/lang/expressions/xml_template_expr.balt [hunk 7]
```diff
@@ -0,0 +1,765 @@
...
+Description: Test with CDATA in xml content.
+Fail-Issue: ballerina-platform/ballerina-lang#33081
+Labels: xml-template-expr, xml
+
+public function init() {
+    int x = 9;
+    int y = 9;
+    xml z = xml `<exampleOfACDATA><![CDATA[${x}character data${y}]]></exampleOfACDATA>`;
+    io:println(z); // @output <exampleOfACDATA><![CDATA[9character data9]]></exampleOfACDATA>
+}
```
Comment (suleka96) [inline conformance/lang/expressions/xml_template_expr.balt:765]: Have we added tests for 

> An xml-template-expr occurring within a const-expr will construct an xml value that has its read-only bit on.

**Review by suleka96 — CHANGES_REQUESTED** [review]

File: conformance/lang/expressions/xml_template_expr.balt (hunk 5 above)
Comment (chiranSachintha) [reply conformance/lang/expressions/xml_template_expr.balt:390]: `module-type-defn` as well.

File: conformance/lang/expressions/xml_template_expr.balt [hunk 8]
```diff
@@ -0,0 +1,765 @@
...
+    io:println(z); // @output <A><B>true</B></A>
+
+    z = xml `<A>${xml `<B>${x === y}</B>`}</A>`;
+    io:println(z); // @output <A><B>true</B></A>
+
+    z = xml `<A>${xml `<B>${x !== s}</B>`}</A>`;
+    io:println(z); // @output <A><B>true</B></A>
+}
+
+Test-Case: output
```
Comment (chiranSachintha) [inline conformance/lang/expressions/xml_template_expr.balt:755]: Shall we add a few tests with type-cast-expr?

File: conformance/lang/expressions/xml_template_expr.balt [hunk 9]
```diff
@@ -0,0 +1,765 @@
+Test-Case: error
+Description: Test xml interpolated expression usage in namespace attribute values.
+Labels: xml-template-expr, xml, raw-template-expr, union-type, float, decimal, string, int
+
+public function init() {
```
Comment (chiranSachintha) [inline conformance/lang/expressions/xml_template_expr.balt:5]: `init` function cannot be public. Need to change every other places

**Pull request closed without merging**